spark/src/test/java/org/apache/iceberg/spark/data/TestSparkOrcUnions.java

```diff
@@ -115,106 +118,28 @@ public void testComplexUnion() throws IOException {
     writer.close();
 
     // Test non-vectorized reader
-    List<InternalRow> internalRows = Lists.newArrayList();
+    List<InternalRow> actualRows = Lists.newArrayList();
     try (CloseableIterable<InternalRow> reader = ORC.read(Files.localInput(orcFile))
         .project(expectedSchema)
         .createReaderFunc(readOrcSchema -> new SparkOrcReader(expectedSchema, readOrcSchema))
         .build()) {
-      reader.forEach(internalRows::add);
+      reader.forEach(actualRows::add);
 
-      Assert.assertEquals(internalRows.size(), NUM_OF_ROWS);
-      assertEquals(expectedSchema, expectedFirstRow, internalRows.get(0));
-      assertEquals(expectedSchema, expectedSecondRow, internalRows.get(1));
+      Assert.assertEquals(actualRows.size(), NUM_OF_ROWS);
+      assertEquals(expectedSchema, expectedFirstRow, actualRows.get(0));
+      assertEquals(expectedSchema, expectedSecondRow, actualRows.get(1));
     }
 
     // Test vectorized reader
-    List<ColumnarBatch> columnarBatches = Lists.newArrayList();
     try (CloseableIterable<ColumnarBatch> reader = ORC.read(Files.localInput(orcFile))
         .project(expectedSchema)
         .createBatchedReaderFunc(readOrcSchema ->
             VectorizedSparkOrcReaders.buildReader(expectedSchema, readOrcSchema, ImmutableMap.of()))
         .build()) {
-      reader.forEach(columnarBatches::add);
-      Iterator<InternalRow> rowIterator = columnarBatches.get(0).rowIterator();
+      final Iterator<InternalRow> actualRowsIt = batchesToRows(reader.iterator());
 
-      Assert.assertEquals(columnarBatches.get(0).numRows(), NUM_OF_ROWS);
-      assertEquals(expectedSchema, expectedFirstRow, rowIterator.next());
-      assertEquals(expectedSchema, expectedSecondRow, rowIterator.next());
-    }
-  }
-
-  @Test
-  public void testSingleComponentUnion() throws IOException {
```

Comment:
Why test for single component union is removed? How is `[ "string" ]` represented in the new schema?

Reply:
We already discussed before we won't support this schema and user should never use single type union cuz it doesn't make sense.

Reply:
Given it is a valid Avro schema, do we have a way to prevent user from creating this kinda of schema in the first place? What is the behavior if user do create it for current implementation?

Reply:
The way to prevent it will be upon the due diligence of the user to not create such schema. We can just tell the user this case is not supported and behavior is undefined.

Reply:
Actually, I have a second thought on this, @autumnust when gobblin writes/converts an ORC dataset and encounters an Avor `[null, string]` type, does it write just `string` type to ORC file, or does it write a union? My intuition is that you guys write a single `string` as ORC schema is nullable by itself.

Reply:
So Gobblin does the transformation: https://github.com/apache/gobblin/blob/5bc22a5502ddeb810a35b0fb996dd9dfc8c81121/gobblin-utility/src/main/java/org/apache/gobblin/util/orc/AvroOrcSchemaConverter.java#L62 

So a default-value representation in Avro shall not leave in a state as single-element union.